### ISSUE TITLE
[Feature Request] Add markdown view of answers

### ISSUE THREAD

**Issue by ahoneybun** [elevenhsoft/cosmic-applet-ollama#1]
This would be helpful for code questions at least from my POV.

**Comment by elevenhsoft** [comment]
Yes, exactly! This is something I was consider before but was not sure how to implement this. I was thinking to use cosmic-text somehow, but maybe markdown parser should be enough as you say. Definitely. So I have a plan for today, thank you ;p

**Comment by elevenhsoft** [comment]
Well, I tried to do this but it's kind of hard task. Especially syntax highlighting for code so I guess this issue will wait a bit for better times :P

**Comment by elevenhsoft** [comment]
Hey @ahoneybun ,
I'm working on this markdown on `markdown` branch. Would you be able to test it and give me some feedback how you see it?

**Comment by elevenhsoft** [comment]
Thank you, I'm working also to add rich_text from cosmic-text, and then we will have some headings, and other stuff like bold text, it will looks better but for now it's kind of bare bone, but it works ^^

**Comment by ahoneybun** [comment]
I think you can say that it works.

![screenshot-2024-06-14-12-42-12](https://github.com/elevenhsoft/cosmic-ext-applet-ollama/assets/4884946/ca146b7e-dae3-49d8-82c5-0843d19f933c)

**Comment by ahoneybun** [comment]
It's odd that it is applying the color outside of the bash code though.

**Comment by elevenhsoft** [comment]
> It's odd that it is applying the color outside of the bash code though.

Yes and I'm not sure how to deal with it. Maybe `rich_text()` will help somehow and I don't know yet.
Syntax is set for whole text block when it recognize code block, and then it works with all the text, that's the result.

After implementing rich text i will try to correct this.

**Comment by ahoneybun** [comment]
Sounds good!

**Comment by elevenhsoft** [comment]
Just updated branch with new update :)

You can check in free time.

**Comment by mmstick** [comment]
There may be value in having a markdown widget integrated into libcosmic

**Comment by elevenhsoft** [comment]
Well, if it's good enough hahah
I'm worry to add it upstream since I'm not sure how good it is. For my use case it works but you know

**Comment by mmstick** [comment]
Code quality can only go up after it's mainstreamed and used by multiple applications.
We could use this in the app store for app summaries. App summaries are written in HTML, which we could pass through https://github.com/nabbisen/mdka-rs to convert to markdown.

**Comment by elevenhsoft** [comment]
Yeah that's for sure. I will try to improve this if possible and make a PR to libcosmic then.
In any case it was nice learning going through all this ;p

**Comment by ahoneybun** [comment]
> Just updated branch with new update :)
> 
> You can check in free time.

It looks like it is still doing it from what I can tell with a prompt like this:

"write a python script to print the weather"

**Comment by elevenhsoft** [comment]
I'm not sure but for me works correctly. Did you wait for full response from AI? The code is properly parsed when full code has been written.

**Comment by ahoneybun** [comment]
@elevenhsoft ah yes that looks correct.

![screenshot-2024-06-18-19-38-13](https://github.com/elevenhsoft/cosmic-ext-applet-ollama/assets/4884946/526fecae-99e4-47ed-b9eb-59afff1d0fd5)
